noir_stdlib/src/scalar_mul.nr

```diff
@@ -12,3 +17,10 @@ pub fn fixed_base_embedded_curve(
 ) -> [Field; 2]
 // docs:end:fixed_base_embedded_curve
 {}
+
+#[foreign(embedded_curve_add)]
+pub fn embedded_curve_add(_point1: EmbeddedCurvePoint, _point2: EmbeddedCurvePoint) -> EmbeddedCurvePoint {}
+
+pub fn embedded_curve_double(point: EmbeddedCurvePoint) -> EmbeddedCurvePoint {
+    embedded_curve_add(point, point)
+}
```

Comment:
@guipublic, thoughts on making these private and then implementing `Add` and a `double` method on `EmbeddedCurvePoint?